microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -302,38 +301,88 @@ def _kdump_test(self, node: Node, log_path: Path, log: Logger) -> None:
             except Exception as identifier:
                 log.debug(
-                    "failed to connect SSH "
+                    "Fail to connect SSH "
                     f"{remote_node._connection_info.address}:"
                     f"{remote_node._connection_info.port}. "
                     f"{identifier.__class__.__name__}: {identifier}. Retry..."
                 )
-                time.sleep(10)
-            if not stdout:
-                serial_console = node.features[SerialConsole]
                 serial_console.check_initramfs(
                     saved_path=log_path, stage="after_trigger_crash", force_run=True
                 )
-        if not stdout:
-            raise LisaException("Timeout to connect the VM after triggering kdump.")
+                system_disconnected = True
+                continue
 
-        # After trigger kdump, the VM will reboot. We need to close and re-initialize
-        node.close()
-        node.initialize()
-        # Verify if the kernel kdump process creates a vmcore file of size 10M+
-        kdump.check_vmcore_exist()
+            # If there is no exception, then the system is connected
+            system_disconnected = False
+
+            # After trigger kdump, the VM will reboot. We need to close and initialize
+            node.close()
+            dump_file = kdump.get_dumpfile_name()
+            saved_dumpfile_size = 0
+            while True:
+                try:
+                    result = node.execute(
+                        f"find {kdump.dump_path} -name {dump_file} -type f -size +10M",
+                        shell=True,
+                        sudo=True,
+                    )
+                    if result.exit_code == 0:
+                        break
+
+                    # When the system is dumping vmcore, but doesn't complete,
+                    # then the vmcore file is named as *-incomplete
+                    result = node.execute(
+                        f"find {kdump.dump_path} -name '*incomplete*'",
+                        shell=True,
+                        sudo=True,
+                    )
+                except Exception as identifier:
+                    log.debug(
+                        "Fail to execute command. It may be caused by the system kernel"
+                        " reboot after dumping vmcore."
+                        f"{identifier.__class__.__name__}: {identifier}. Retry..."
+                    )
+                    system_disconnected = True
+                    break
+                # Getting tool here can avoiding failure caused by system disconnected
+                stat = node.tools[Stat]
+                if result.exit_code == 0:
+                    incomplete_file = result.stdout
+                    incomplete_file_size = stat.get_total_size(incomplete_file)
+                    if incomplete_file_size > saved_dumpfile_size:
+                        saved_dumpfile_size = incomplete_file_size
```

Comment:
does it mean when vmcore is incomplete, case still pass?

Reply:
No, if so, and it hasn't achieved the timeout time, it will still be in this loop, and re-try to get the vmcore file.